wnfs-common/src/utils/test.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{encode, BlockStore, MemoryBlockStore, CODEC_DAG_CBOR, CODEC_RAW};
+use anyhow::Result;
+use base64_serde::base64_serde_type;
+use bytes::Bytes;
+use libipld::{
+    cbor::DagCborCodec,
+    json::DagJsonCodec,
+    prelude::{Decode, Encode},
+    Cid, Ipld,
+};
+use proptest::{
+    strategy::{Strategy, ValueTree},
+    test_runner::TestRunner,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{
+    collections::{BTreeMap, HashMap},
+    io::Cursor,
+};
+
+//--------------------------------------------------------------------------------------------------
+// Type Definitions
+//--------------------------------------------------------------------------------------------------
+
+type BlockHandler = Box<dyn Fn(&Bytes) -> Result<Bytes>>;
+
+#[derive(Default)]
+pub struct SnapshotBlockStore {
+    inner: MemoryBlockStore,
+    block_handlers: HashMap<Cid, BlockHandler>,
+}
+
+base64_serde_type!(Base64Standard, base64::engine::general_purpose::STANDARD);
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BlockSnapshot {
+    pub value: Value,
+    #[serde(with = "Base64Standard")]
+    pub cbor: Vec<u8>,
```

Comment:
Sorry, just one more minor nit, can we call this "bytes", now that sometimes it's not CBOR, but e.g. encrypted bytes:
```suggestion
    pub bytes: Vec<u8>,
```